src/actions/base.ts

```diff
@@ -83,7 +83,9 @@ export abstract class BaseAction {
     }
 
     const keys2D = BaseAction.is2DArray(this.keys) ? this.keys : [this.keys];
-    const keysSlice = keys2D.map((x) => x.slice(0, keysPressed.length));
+    const keysSlice = keys2D.map((x: string[] | readonly string[]) =>
```

Comment:
Why is this necessary?

Reply:
The build was failing (typescript compilation stage) without this change.

Reply:
Weird. I just checked and it built fine, so reverting this line to keep the change contained